### DISCUSSION TITLE
SharpPcap BufferSize / KernelBufferSize parameters versus GarbageCollection in C#

### DISCUSSION THREAD

**Discussion by pizzdov** [chmorgan/sharppcap#535] [Q&A]
Is there any way to programmatically increase the BufferSize or KernelBufferSize parameters?

When processing large volumes of network packets in C#, I have noticed packets being dropped while GarbageCollection is active.  I was hoping to be able to increase the low level packet buffers to help increase the probabilty of capturing these dropped packets.

**Comment by kayoub5** [comment]
You can specify them in the Open function configuration object.

You can't change the buffer sizes once the device is opened.

**Reply by pizzdov** [reply]
Thanks.  I see it now.  Passing in a DeviceConfiguration.

Do you know what the default size of the buffers are?

**Reply by kayoub5** [reply]
If you don't provide one to sharppcap, libpcap uses its own default buffer size. 

I think 2MB, cross check with libpcap developers to be sure
